volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,505 @@
+
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(name="kernel", description="Linux kernel",
+                                           architectures=["Intel32", "Intel64"]),
+            requirements.VersionRequirement(name="SockHandlers", component=SockHandlers, version=(1, 0, 0)),
+            requirements.PluginRequirement(name="lsof", plugin=lsof.Lsof, version=(1, 1, 0)),
+            requirements.VersionRequirement(name="linuxutils", component=linux.LinuxUtilities, version=(2, 0, 0)),
+            requirements.BooleanRequirement(name="unix",
+                                            description=("Show UNIX domain Sockets only"),
+                                            default=False,
+                                            optional=True),
+            requirements.ListRequirement(name="pids",
+                                         description="Filter results by process IDs. "
+                                                     "It takes the root PID namespace identifiers.",
+                                         element_type=int,
+                                         optional=True),
+            requirements.IntRequirement(name="netns",
+                                        description="Filter results by network namespace. "
+                                                    "Otherwise, all of them are shown.",
+                                        optional=True),
+        ]
+
+    @classmethod
+    def list_sockets(cls,
+                     context: interfaces.context.ContextInterface,
+                     symbol_table: str,
+                     filter_func: Callable[[int], bool] = lambda _: False):
+        """Returns every single socket descriptor
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            symbol_table: The name of the kernel module on which to operate
+            filter_func: A function which takes a task object and returns True if the task should be ignored/filtered
+
+        Yields:
+            task: Kernel's task object
+            netns_id: Network namespace ID
+            fd_num: File descriptor number
+            family: Socket family string (AF_UNIX, AF_INET, etc)
+            sock_type: Socket type string (STREAM, DGRAM, etc)
+            protocol: Protocol string (UDP, TCP, etc)
+            sock_fields: A tuple with the *_sock object, the sock stats and the
+                         extended info dictionary
+        """
+        vmlinux = context.modules[symbol_table]
+
+        sfop_addr = vmlinux.object_from_symbol("socket_file_ops").vol.offset
+        dfop_addr = vmlinux.object_from_symbol("sockfs_dentry_operations").vol.offset
+
+        fd_generator = lsof.Lsof.list_fds(context, vmlinux.name, filter_func)
+        for _pid, _task_comm, task, fd_fields in fd_generator:
```

Comment:
@ikelos for '_pid' and '_task_comm' that are not used, can we use the normal convention of just '_' or do we need the prefixed name?

Reply:
@ikelos bump on this comment

Reply:
Err, you could use just `_`, but particularly when there are several being ignored, I find it better to just put the name behind the `_` so people referring to the code can tell what was hidden and what to expect if they need to access it.  I'd leave these as if you'd be ok with that?

Reply:
sounds good to me. It was just a style question so the codebase would be consistent.